File: 401 to 500/409_longest-palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        count_alpha = {}
        for i in s:
            if i not in count_alpha:
                count_alpha[i] = 1
            else:
                count_alpha[i] += 1
        c = []
        for i in count_alpha:
            c.append(count_alpha[i])
        c.sort(reverse = True)
        out = 0
        odd = False
        for val in c:
            if val % 2 == 0:
                out += val
            else:
                out += val-1
                odd = True
        if odd:
            out += 1
        return out

File: 401 to 500/C++/409_longest-palindrome.cpp
#include<bits/stdc++.h>
using namespace std;
#define endl '\n'
#define int long long
const int mod = 1e9+7;

class Solution {
public:
    int longestPalindrome(string s) {
        unordered_map<char,int> m;
        for(auto x:s) m[x]++;
        int len = 0;
        bool odd = false;
        for(auto x : m){
            int val = x.second;
            if(val % 2 == 0) len += val;
            else len += val -1, odd = true;
        }
        return odd ? len + 1 : len;
    }
};

signed main(){
    ios_base::sync_with_stdio(false);
    cin.tie(0);cout.tie(0);
    Solution obj;
}